src/vcpkg/binarycaching.cpp

```diff
@@ -801,36 +801,24 @@ namespace
         {
         }
 
-        Command command() const
-        {
-            Command cmd;
-            cmd.string_arg("curl")
-                .string_arg("-s")
-                .string_arg("-H")
-                .string_arg("Content-Type: application/json")
-                .string_arg("-H")
-                .string_arg(m_token_header)
-                .string_arg("-H")
-                .string_arg(m_accept_header);
-            return cmd;
-        }
-
-        std::string lookup_cache_entry(const std::string& abi) const
+        std::string lookup_cache_entry(StringView name, const std::string& abi) const
         {
-            auto cmd = command()
-                           .string_arg(m_url)
-                           .string_arg("-G")
-                           .string_arg("-d")
-                           .string_arg("keys=vcpkg")
-                           .string_arg("-d")
-                           .string_arg("version=" + abi);
-
-            std::vector<std::string> lines;
-            auto res = cmd_execute_and_capture_output(cmd);
-            if (!res.has_value() || res.get()->exit_code) return {};
-            auto json = Json::parse_object(res.get()->output);
-            if (!json.has_value() || !json.get()->contains("archiveLocation")) return {};
-            return json.get()->get("archiveLocation")->string(VCPKG_LINE_INFO).to_string();
+            auto url = format_url_query(m_url, std::vector<std::string>{"keys=" + name + "-" + abi, "version=" + abi});
+            auto res =
+                invoke_http_request("GET",
+                                    std::vector<std::string>{
+                                        m_content_type_header.to_string(), m_token_header, m_accept_header.to_string()},
+                                    url);
+            auto maybe_json = Json::parse_object(res.get()->c_str());
```

Comment:
I think we need to check `res.get()` here

Reply:
Alternately, you can use the nested if form you used on 886; I slightly prefer 886 over my suggestion. (The suggestion just fit on one line :) )